docs/contributing/database-migrations/mssql.md

```diff
@@ -0,0 +1,109 @@
+using.
+
+=== "Windows" Visual Studio provides built in support for Database projects with their SQL Server
+Data Tools. This is usually the optimal development environment with built in support for schema
+comparison and much more.
+
+=== "Mac & Linux" Visual Studio for Mac unfortunately does not support Database projects, however
+you may instead use [Visual Studio Code](vscode) or [Azure Data Studio](azureds) with the
+[SQL Database Projects](SDPE) extension, which provides schema comparison and more. You may also
+modify the `.sql` files directly with any text editor.
+
+    Do note that when adding or renaming SQL files you might need to manually update the references
+    in the `.sqlproj` file.
+
+To make a database change, start by modifying the `.sql` files in `src/Sql/dbo`. These changes will
+also need to be applied in a migration script. Migration scripts are located in
+`util/Migrator/DbScripts`.
+
+You can either generate the migration scripts automatically using the _Schema Comparison_
+functionality or by manually writing them. Do note that the automatic method will only take you so
+far and it will need to be manually edited to adhere to the code styles.
+
+## Modifying the database
+
+Since we follow [Evolutionary Database Design _(EDD)_](./edd.mdx), any migration that modifies
+existing columns most likely needs to be split into at least two parts: a backwards compatible
+transition phase, and a non-backwards compatible phase.
+
+### Best Practices
+
+When writing a migration script there are a couple of best practices we follow. Please check the
+[T-SQL Code Style][code-style-sql] for more details. But the most important aspect is ensuring the
+script can be re-run on the database multiple times without producing any errors or data loss.
+
+### Backwards Compatible
+
+Since we follow _EDD_ the first migration needs to retain backwards compatibility with existing
+production code.
+
+1. Modify the source `.sql` files in `src/Sql/dbo`.
+2. Write a migration script, and place it in `util/Migrator/DbScripts`. Each script _must_ be
+   prefixed with the current date.
+
+Please take care to ensure any existing _Stored Procedure_ accepts the same input parameters which
+ensures backwards compatibility. In the case a column is renamed, moved care needs to be taken to
+ensure the existing sprocs first checks the new location before falling back to the old location. We
+also need to ensure we continue updating the old data columns, since in case a rollback is necessary
+no data should be lost.
+
+### Data Migration
+
+We now need to write a script that migrates any data from the old location to the new locations.
+This script should ideally be written in a way that supports batching, i.e. execute for X number of
+rows at a time. This helps avoiding locking the database. When running the scripts against the
+server please keep running it until it affects `0 rows`.
```

Comment:
Need to decide if data migrations are destructive or non destructive. I.e. duplicate the data for a while.